In [59]:
# Importing packages
import pandas as pd
import numpy as np

#Setting up the pandas to see all the columns - will take longer but very convenient
pd.set_option('display.max_columns', None)

# Load Dataset
dataset = pd.read_csv('~/Desktop/Speed Dating Experiment/Data/Speed Dating Data.csv', encoding="ISO-8859-1")

First, we're going to take a quick look at what we are working for, and potentially clean up a bit

In [60]:
# Taking a first look at the dataset
dataset.head()

iid   id  gender  idg  condtn  wave  round  position  positin1  order  \
0    1  1.0       0    1       1     1     10         7       NaN      4   
1    1  1.0       0    1       1     1     10         7       NaN      3   
2    1  1.0       0    1       1     1     10         7       NaN     10   
3    1  1.0       0    1       1     1     10         7       NaN      5   
4    1  1.0       0    1       1     1     10         7       NaN      7   

   partner   pid  match  int_corr  samerace  age_o  race_o  pf_o_att  \
0        1  11.0      0      0.14         0   27.0     2.0      35.0   
1        2  12.0      0      0.54         0   22.0     2.0      60.0   
2        3  13.0      1      0.16         1   22.0     4.0      19.0   
3        4  14.0      1      0.61         0   23.0     2.0      30.0   
4        5  15.0      1      0.21         0   24.0     3.0      30.0   

   pf_o_sin  pf_o_int  pf_o_fun  pf_o_amb  pf_o_sha  dec_o  attr_o  sinc_o  \
0      20.0      20.0      20.0       0.0       5.0      0     6.0     8.0   
1       0.0       0.0      40.0       0.0       0.0      0     7.0     8.0   
2      18.0      19.0      18.0      14.0      12.0      1    10.0    10.0   
3       5.0      15.0      40.0       5.0       5.0      1     7.0     8.0   
4      10.0      20.0      10.0      10.0      20.0      1     8.0     7.0   

   intel_o  fun_o  amb_o  shar_o  like_o  prob_o  met_o   age field  field_cd  \
0      8.0    8.0    8.0     6.0     7.0     4.0    2.0  21.0   Law       1.0   
1     10.0    7.0    7.0     5.0     8.0     4.0    2.0  21.0   Law       1.0   
2     10.0   10.0   10.0    10.0    10.0    10.0    1.0  21.0   Law       1.0   
3      9.0    8.0    9.0     8.0     7.0     7.0    2.0  21.0   Law       1.0   
4      9.0    6.0    9.0     7.0     8.0     6.0    2.0  21.0   Law       1.0   

  undergra mn_sat tuition  race  imprace  imprelig     from zipcode  \
0      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
1      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
2      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
3      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
4      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   

      income  goal  date  go_out  career  career_c  sports  tvsports  \
0  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
1  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
2  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
3  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
4  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   

   exercise  dining  museums  art  hiking  gaming  clubbing  reading   tv  \
0       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
1       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
2       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
3       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
4       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   

   theater  movies  concerts  music  shopping  yoga  exphappy  expnum  \
0      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
1      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
2      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
3      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
4      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   

   attr1_1  sinc1_1  intel1_1  fun1_1  amb1_1  shar1_1  attr4_1  sinc4_1  \
0     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
1     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
2     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
3     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
4     15.0     

In [61]:
#Because we are displaying all of the rows, we're going to be careful about not displaying more than needed. 
#For those type of data, the average and distribution data is not particularly interesting to us, we will only display the number of non-null data (count) and the min and max to see what the data ranges between
dataset.describe().loc[['count','min','max']] 

iid      id  gender     idg  condtn    wave   round  position  \
count  8378.0  8377.0  8378.0  8378.0  8378.0  8378.0  8378.0    8378.0   
min       1.0     1.0     0.0     1.0     1.0     1.0     5.0       1.0   
max     552.0    22.0     1.0    44.0     2.0    21.0    22.0      22.0   

       positin1   order  partner     pid   match  int_corr  samerace   age_o  \
count    6532.0  8378.0   8378.0  8368.0  8378.0   8220.00    8378.0  8274.0   
min         1.0     1.0      1.0     1.0     0.0     -0.83       0.0    18.0   
max        22.0    22.0     22.0   552.0     1.0      0.91       1.0    55.0   

       race_o  pf_o_att  pf_o_sin  pf_o_int  pf_o_fun  pf_o_amb  pf_o_sha  \
count  8305.0    8289.0    8289.0    8289.0    8280.0    8271.0    8249.0   
min       1.0       0.0       0.0       0.0       0.0       0.0       0.0   
max       6.0     100.0      60.0      50.0      50.0      53.0      30.0   

        dec_o  attr_o  sinc_o  intel_o   fun_o   amb_o  shar_o  like_o  \
count  8378.0  8166.0  8091.0   8072.0  8018.0  7656.0  7302.0  8128.0   
min       0.0     0.0     0.0      0.0     0.0     0.0     0.0     0.0   
max       1.0    10.5    10.0     10.0    11.0    10.0    10.0    10.0   

       prob_o   met_o     age  field_cd    race  imprace  imprelig    goal  \
count  8060.0  7993.0  8283.0    8296.0  8315.0   8299.0    8299.0  8299.0   
min       0.0     1.0    18.0       1.0     1.0      0.0       1.0     1.0   
max      10.0     8.0    55.0      18.0     6.0     10.0      10.0     6.0   

         date  go_out  career_c  sports  tvsports  exercise  dining  museums  \
count  8281.0  8299.0    8240.0  8299.0    8299.0    8299.0  8299.0   8299.0   
min       1.0     1.0       1.0     1.0       1.0       1.0     1.0      0.0   
max       7.0     7.0      17.0    10.0      10.0      10.0    10.0     10.0   

          art  hiking  gaming  clubbing  reading      tv  theater  movies  \
count  8299.0  8299.0  8299.0    8299.0   8299.0  8299.0   8299.0  8299.0   
min       0.0     0.0     0.0       0.0      1.0     1.0      0.0     0.0   
max      10.0    10.0    14.0      10.0     13.0    10.0     10.0    10.0   

       concerts   music  shopping    yoga  exphappy  expnum  attr1_1  sinc1_1  \
count    8299.0  8299.0    8299.0  8299.0    8277.0  1800.0   8299.0   8299.0   
min         0.0     1.0       1.0     0.0       1.0     0.0      0.0      0.0   
max        10.0    10.0      10.0    10.0      10.0    20.0    100.0     60.0   

       intel1_1  fun1_1  amb1_1  shar1_1  attr4_1  sinc4_1  intel4_1  fun4_1  \
count    8299.0  8289.0  8279.0   8257.0   6489.0   6489.0    6489.0  6489.0   
min         0.0     0.0     0.0      0.0      5.0      0.0       0.0     0.0   
max        50.0    50.0    53.0     30.0     95.0     35.0      35.0    45.0   

       amb4_1  shar4_1  attr2_1  sinc2_1  intel2_1  fun2_1  amb2_1  shar2_1  \
count  6489.0   6467.0   8299.0   8299.0    8299.0  8299.0  8289.0   8289.0   
min       0.0      0.0      0.0      0.0       0.0     0.0     0.0      0.0   
max      50.0     40.0    100.0     50.0      40.0    50.0    50.0     30.0   

       attr3_1  sinc3_1  fun3_1  intel3_1  amb3_1  attr5_1  sinc5_1  intel5_1  \
count   8273.0   8273.0  8273.0    8273.0  8273.0   4906.0   4906.0    4906.0   
min        2.0      2.0     2.0       3.0     2.0      2.0      1.0       3.0   
max       10.0     10.0    10.0      10.0    10.0     10.0     10.0      10.0   

       fun5_1  amb5_1     dec    attr    sinc   intel     fun     amb    shar  \
count  4906.0  4906.0  8378.0  8176.0  8101.0  8082.0  8028.0  7666.0  7311.0   
min       2.0     1.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
max      10.0    10.0     1.0    10.0    10.0    10.0    10.0    10.0    10.0   

         like    prob     met  match_es  attr1_s  sinc1_s  intel1_s  fun1_s  \
count  8138.0  8069.0  8003.0    7205.0   4096.0   4096.0    4096.0  4096.0   
min       0.0     0.0     0.0       0.0      3.0      0.0       0.0 

In [63]:
print("The number of unique participant id is",dataset['iid'].nunique())

for a, b in zip(np.arange(1,553), dataset['iid'].unique()):
        if a != b: #Here we are essentially comparing our list of participants i.d. to a normal range from 1 to 552 to print where both array stop beeing equal, thus the missing participant i.d.
            print("The missing participant i.d. is {}".format(a))
            break #We break here because every number thereafter is going to be different and we don't need them to all be calculated/ printed

The number of unique participant id is 551
The missing participant i.d. is 118


<br>
<br>
Now, we're noticing quite a few interesting things here, namely: <br>
    * While participants i.d. go up to number 552, there is 551 participants in reality <br>
    * Only two genders in this data are included: Female and Male <br>
    * Participants' age ranged between 18 years old and 55 years old. Would be interesting to see the age difference between the matches? <br>

<br>
<br>

In [9]:
dataset[['iid','id']].transpose()

0     1     2     3     4     5     6     7     8     9     10    11    \
iid   1.0   1.0   1.0   1.0   1.0   1.0   1.0   1.0   1.0   1.0   2.0   2.0   
id    1.0   1.0   1.0   1.0   1.0   1.0   1.0   1.0   1.0   1.0   2.0   2.0   

     12    13    14    15    16    17    18    19    20    21    22    23    \
iid   2.0   2.0   2.0   2.0   2.0   2.0   2.0   2.0   3.0   3.0   3.0   3.0   
id    2.0   2.0   2.0   2.0   2.0   2.0   2.0   2.0   3.0   3.0   3.0   3.0   

     24    25    26    27    28    29    30    31    32    33    34    35    \
iid   3.0   3.0   3.0   3.0   3.0   3.0   4.0   4.0   4.0   4.0   4.0   4.0   
id    3.0   3.0   3.0   3.0   3.0   3.0   4.0   4.0   4.0   4.0   4.0   4.0   

     36    37    38    39    40    41    42    43    44    45    46    47    \
iid   4.0   4.0   4.0   4.0   5.0   5.0   5.0   5.0   5.0   5.0   5.0   5.0   
id    4.0   4.0   4.0   4.0   5.0   5.0   5.0   5.0   5.0   5.0   5.0   5.0   

     48    49    50    51    52    53    54    55    56    57    58    59    \
iid   5.0   5.0   6.0   6.0   6.0   6.0   6.0   6.0   6.0   6.0   6.0   6.0   
id    5.0   5.0   6.0   6.0   6.0   6.0   6.0   6.0   6.0   6.0   6.0   6.0   

     60    61    62    63    64    65    66    67    68    69    70    71    \
iid   7.0   7.0   7.0   7.0   7.0   7.0   7.0   7.0   7.0   7.0   8.0   8.0   
id    7.0   7.0   7.0   7.0   7.0   7.0   7.0   7.0   7.0   7.0   8.0   8.0   

     72    73    74    75    76    77    78    79    80    81    82    83    \
iid   8.0   8.0   8.0   8.0   8.0   8.0   8.0   8.0   9.0   9.0   9.0   9.0   
id    8.0   8.0   8.0   8.0   8.0   8.0   8.0   8.0   9.0   9.0   9.0   9.0   

     84    85    86    87    88    89    90    91    92    93    94    95    \
iid   9.0   9.0   9.0   9.0   9.0   9.0  10.0  10.0  10.0  10.0  10.0  10.0   
id    9.0   9.0   9.0   9.0   9.0   9.0  10.0  10.0  10.0  10.0  10.0  10.0   

     96    97    98    99    100   101   102   103   104   105   106   107   \
iid  10.0  10.0  10.0  10.0  11.0  11.0  11.0  11.0  11.0  11.0  11.0  11.0   
id   10.0  10.0  10.0  10.0   1.0   1.0   1.0   1.0   1.0   1.0   1.0   1.0   

     108   109   110   111   112   113   114   115   116   117   118   119   \
iid  11.0  11.0  12.0  12.0  12.0  12.0  12.0  12.0  12.0  12.0  12.0  12.0   
id    1.0   1.0   2.0   2.0   2.0   2.0   2.0   2.0   2.0   2.0   2.0   2.0   

     120   121   122   123   124   125   126   127   128   129   130   131   \
iid  13.0  13.0  13.0  13.0  13.0  13.0  13.0  13.0  13.0  13.0  14.0  14.0   
id    3.0   3.0   3.0   3.0   3.0   3.0   3.0   3.0   3.0   3.0   4.0   4.0   

     132   133   134   135   136   137   138   139   140   141   142   143   \
iid  14.0  14.0  14.0  14.0  14.0  14.0  14.0  14.0  15.0  15.0  15.0  15.0   
id    4.0   4.0   4.0   4.0   4.0   4.0   4.0   4.0   5.0   5.0   5.0   5.0   

     144   145   146   147   148   149   150   151   152   153   154   155   \
iid  15.0  15.0  15.0  15.0  15.0  15.0  16.0  16.0  16.0  16.0  16.0  16.0   
id    5.0   5.0   5.0   5.0   5.0   5.0   6.0   6.0   6.0   6.0   6.0   6.0   

     156   157   158   159   160   161   162   163   164   165   166   167   \
iid  16.0  16.0  16.0  16.0  17.0  17.0  17.0  17.0  17.0  17.0  17.0  17.0   
id    6.0   6.0   6.0   6.0   7.0   7.0   7.0   7.0   7.0   7.0   7.0   7.0   

     168   169   170   171   172   173   174   175   176   177   178   179   \
iid  17.0  17.0  18.0  18.0  18.0  18.0  18.0  18.0  18.0  18.0  18.0  18.0   
id    7.0   7.0   8.0   8.0   8.0   8.0   8.0   8.0   8.0   8.0   8.0   8.0   

     180   181   182   183   184   185   186   187   188   189   190   191   \
iid  19.0  19.0  19.0  19.0  19.0  19.0  19.0  19.0  19.0  19.0  20.0  20.0   
id    9.0   9.0   9.0   9.0   9.0   9.0   9.0   9.0   9.0   9.0  10.0  10.0   

     192   193   194   195   196   197   198   199   200   201   202   203   \
iid  20.0  20.0  20.0  20.0  20.0  20.0  20.0  20.0  21.0  21.0  21.0  21.0   
id   10.0  10.0  10.0  10.0  10.0  10.0

In [19]:
dataset['id'].describe()

count    8377.000000
mean        8.960248
std         5.491329
min         1.000000
25%         4.000000
50%         8.000000
75%        13.000000
max        22.000000
Name: id, dtype: float64

In [26]:
print(dataset['iid'])

0         1
1         1
2         1
3         1
4         1
       ... 
8373    552
8374    552
8375    552
8376    552
8377    552
Name: iid, Length: 8378, dtype: int64


In [23]:
dataset.loc[dataset['id'] == 1].shape

(552, 195)

In [27]:
dataset['iid'].unique()

array([  1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,  13,
        14,  15,  16,  17,  18,  19,  20,  21,  22,  23,  24,  25,  26,
        27,  28,  29,  30,  31,  32,  33,  34,  35,  36,  37,  38,  39,
        40,  41,  42,  43,  44,  45,  46,  47,  48,  49,  50,  51,  52,
        53,  54,  55,  56,  57,  58,  59,  60,  61,  62,  63,  64,  65,
        66,  67,  68,  69,  70,  71,  72,  73,  74,  75,  76,  77,  78,
        79,  80,  81,  82,  83,  84,  85,  86,  87,  88,  89,  90,  91,
        92,  93,  94,  95,  96,  97,  98,  99, 100, 101, 102, 103, 104,
       105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117,
       119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131,
       132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144,
       145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157,
       158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170,
       171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 18

In [35]:
test

array([  1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,  13,
        14,  15,  16,  17,  18,  19,  20,  21,  22,  23,  24,  25,  26,
        27,  28,  29,  30,  31,  32,  33,  34,  35,  36,  37,  38,  39,
        40,  41,  42,  43,  44,  45,  46,  47,  48,  49,  50,  51,  52,
        53,  54,  55,  56,  57,  58,  59,  60,  61,  62,  63,  64,  65,
        66,  67,  68,  69,  70,  71,  72,  73,  74,  75,  76,  77,  78,
        79,  80,  81,  82,  83,  84,  85,  86,  87,  88,  89,  90,  91,
        92,  93,  94,  95,  96,  97,  98,  99, 100, 101, 102, 103, 104,
       105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117,
       118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130,
       131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143,
       144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156,
       157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169,
       170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 18

In [46]:
test = np.arange(1,553)
list_check = dataset['iid'].unique()


        
        
for a, b in zip(test, list_check):
        if a != b:
            print(a,b)
            break

118 119
